File: src/beginner/day04/Question01.java
package beginner.day04;

public class Question01 {

    public static class Solution {

        public int solution(int n) {
            int answer = 0;

            for (int i = 0; i <= 15; i++) {
                if (n <= i * 7) {
                    answer = i;
                    break;
                }
            }
            
            return answer;
        }
    }

    public static void main(String[] args) {

        Solution solution = new Solution();

        System.out.println(solution.solution(7));
        System.out.println(solution.solution(1));
        System.out.println(solution.solution(15));
    }
}

File: src/beginner/day22/Question01.kt
package beginner.day22

class Question01{

    fun solution(n: Int): Int {
        var answer: Int = 0

        val map = mutableMapOf<Int,Int>()

        for (key in 1..100) {

            while (key <= 100) {

                answer++

                if (answer.toString().contains("3")
                    || answer % 3 == 0) {
                    continue
                }

                map[key] = answer
                break
            }

            if (key == n) {
                answer = map[key]!!
                break
            }
        }

        return answer
    }
}

fun main() {

    val solution = Question01()

    println(solution.solution(15))
    println(solution.solution(40))
}